### PR TITLE
Moved js rules test files to rjson extension.

### PR DESCRIPTION
**Pull request by ematipico**
Implements #728 

Fixed also some edge cases during the creation of the pages and files.

I managed to move almost all the rules. The only rule that is causing troubles is `noReferenceToNonExistingGroup.test.ts`.

For some reason, when converted to rjson and one of the strings is given to `parseJS`, the final AST is different and the expected node that is checked inside the rule is not there.

### CODE REVIEW

File: packages/@romefrontend/compiler/lint/rules/harness.test.ts [hunk 1]
```diff
@@ -5,20 +5,21 @@ import {DiagnosticCategory} from "@romefrontend/diagnostics";
...
 
 	test(
 		name,
 		async (t) => {
 			await testLint(
 				t,
 				{
-					invalid: invalid.map((str) => dedent(str)),
-					valid: valid.map((str) => dedent(str)),
+					invalid: invalid ? invalid.map((str) => dedent(str)) : [],
```
Comment (ematipico) [inline packages/@romefrontend/compiler/lint/rules/harness.test.ts:16]: We have rules where `invalid` and `valid` are empty

File: packages/@romefrontend/compiler/lint/rules/harness.test.ts [hunk 2]
```diff
@@ -5,20 +5,21 @@ import {DiagnosticCategory} from "@romefrontend/diagnostics";
...
+					valid: valid ? valid.map((str) => dedent(str)) : [],
 				},
 				{
 					category: (`lint/${name}` as DiagnosticCategory),
 					snapshotFilename: `${name}.test.md`,
+					path,
```
Comment (ematipico) [inline packages/@romefrontend/compiler/lint/rules/harness.test.ts:22]: Some rules have `path` so I added it

File: scripts/generated-files/lint-rules.cjs [hunk 3]
```diff
@@ -163,17 +163,21 @@ for (const {docs, ruleName, hasRJSON} of defs) {
 	file += "## Examples\n";
 
 	file += "## Invalid\n";
-	for (const example of json.invalid) {
-		file += "```" + language + "\n";
-		file += dedent(example).trim() + "\n";
-		file += "```\n";
+	if (json.invalid) {
```
Comment (ematipico) [inline scripts/generated-files/lint-rules.cjs:166]: Some edge cases where `invalid` and `valid` could be empty. We should not have these cases but I think we can review them later

File: scripts/_utils.cjs [hunk 4]
```diff
@@ -228,13 +228,13 @@ exports.toCamelCase = function(str, forceCapitalize) {
 
 exports.dedent = function(str) {
 	// Find min indentation
-	const match = str.match(/^[ \t]*(?=\S)/gm);
+	const match = String(str).match(/^[ \t]*(?=\S)/gm);
```
Comment (ematipico) [inline scripts/_utils.cjs:231]: Some edge case where `str` was a number

File: packages/@romefrontend/string-utils/dedent.ts [hunk 5]
```diff
@@ -16,11 +16,13 @@ export function dedent(
 	} else {
 		const parts: Array<string> = [];
 
-		// Perform the interpolation
-		for (let i = 0; i < literals.raw.length; i++) {
-			parts.push(literals.raw[i]);
-			if (i < values.length) {
-				parts.push(values[i]);
+		if (literals.raw) {
```
Comment (ematipico) [inline packages/@romefrontend/string-utils/dedent.ts:19]: Some edge case where `literals.raw` was undefined

**Comment by sebmck** [comment]
Woah, awesome! Didn't think we'd get around to having this anytime soon. Happy that we'll have at least some basic docs for all the rules for launch.
